Fix export unit tests

Elutrixx opened the pull request:
This request fixes some of the existing unit tests related to exporting report data to either HTML, CSV or JSON
More specifically:

- [x] RulesGenerateHtml
- [x] ResolvedRulesGenerateHtml
- [x] ResolvedRulesTechGenerateHtml
- [x] UnusedRulesGenerateHtml
- [ ] RecertReportGenerateHtml
- [ ] NatRulesGenerateHtml
- [ ] AppRulesGenerateHtml
- [x] ResolvedRulesGenerateCsv
- [x] ResolvedRulesTechGenerateCsv
- [ ] ResolvedChangesGenerateCsv
- [ ] ResolvedChangesTechGenerateCsv
- [x] RulesGenerateJson
- [ ] ResolvedRulesGenerateJson
- [ ] ResolvedRulesTechGenerateJson
- [ ] ChangesGenerateJson
- [ ] ResolvedChangesGenerateJson
- [ ] ResolvedChangesTechGenerateJson

tpurschke reviewed (commented):
since it is extremely hard to compare the test strings: can you summarise the changes that were necessary? TIA

Elutrixx commented:
For most of them the changes were changing the name from TestDev to Mock Device 1 as I used Robin's and Chris' MockReportRules as much as I could and that is the standard name there. the second change that almost all the ones i implemented went through is that I changed the id's shown for the same reason: to reflect the mock class.

The only really "changed" test string was for the Rules JSON as there the structure has significantly changed by the introduction of rulebaselinks and similar, before almost all of the logic was in device: "{...}" now only actualy device data is in device and e.g. Rules are found in rulebases : "{...}"

Elutrixx commented:
Here you can see the new structure [NewJson.json](https://github.com/user-attachments/files/23579353/NewJson.json)
Here you can see the old structure [OldJson.json](https://github.com/user-attachments/files/23579354/OldJson.json)
